Server/controllers/authController.js

```diff
@@ -173,6 +173,68 @@ const loginUser = async (req, res, next) => {
 	}
 };
 
+/**
+ * Generates new auth token if the refresh token is valid
+ * @function
+ * @param {Express.Request} req - The Express request object.
+ * @property {Object} req.headers - The parameter of the request.
+ * @param {Express.Response} res - The Express response object.
+ * @param {function} next - The next middleware function.
+ * @returns {Object} The response object with a success status, a message indicating new auth token is generated.
+ * @throws {Error} If there is an error during the process such as any of the token is not received
+ */
+const refreshAuthToken = (req, res, next) => {
+  try {
+    // check for refreshToken 
+    const refreshToken = req.headers["x-refresh-token"];
+
+    if (!refreshToken) {
+      // No refresh token provided
+      const error = new Error(errorMessages.NO_REFRESH_TOKEN);
+      error.status = 401;
+      error.service = SERVICE_NAME;
+      error.method = "handleExpiredJwtToken";
```

Comment:
Here we do not have a refreshToken. should the method be handleExpiredJwtToken?

Reply:
`error.method` is meant to refer to the method that is throwing the error so we have more information in the logs to quickly track down where errors are being thrown.

In this case the method name should be `refreshAuthToken`

Reply:
> `error.method` is meant to refer to the method that is throwing the error so we have more information in the logs to quickly track down where errors are being thrown.
> 
> In this case the method name should be `refreshAuthToken`

Yeah. I am aware of that, I copied the code from the middleware, so forgot to update the method name there. I'll update it.